lib/puppet/provider/mysql.rb

```diff
@@ -13,7 +14,22 @@ def self.defaults_file
       nil
     end
   end
-  
+
+  def self.mysqld_version
+    # we cache the result ...
+    # note: be prepared for '5.7.6-rc-log' etc results
+    #       versioncmp detects 5.7.6-log to be newer then 5.7.6
+    #       this is why we need the trimming.
+    return @mysqld_version_string.scan(/\d+\.\d+\.\d+/).first unless @mysqld_version_string.nil?
```

Comment:
Shouldn't this regex (and the one below) be anchored with `^`, or at least have an additional space in front to not accidentally match something else?